### DISCUSSION TITLE
Failed to load a simple project

### DISCUSSION THREAD

**Discussion by hmilyyang** [nxrighthere/UnrealCLR#166] [Help] [answered]
I followed the manual page and created a simple .NET class library project. 
It compiled OK. But when I click Run in Unreal Engine，some error text displayed:

LogUnrealCLR: Error: UnrealCLR::Module::Exception: Loading of assemblies failed
System.BadImageFormatException: Could not load file or assembly 'TestUnrealClrLib, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'. Reference assemblies should not be loaded for execution.  They can only be loaded in the Reflection-only loader context. (0x80131058)
File name: 'TestUnrealClrLib, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' ---> System.BadImageFormatException: Cannot load a reference assembly for execution.
   at System.Runtime.Loader.AssemblyLoadContext.InternalLoad(ReadOnlySpan`1 arrAssembly, ReadOnlySpan`1 arrSymbols)
   at System.Runtime.Loader.AssemblyLoadContext.LoadFromStream(Stream assembly, Stream assemblySymbols)
   at UnrealEngine.Plugins.Loader.ManagedLoadContext.LoadAssemblyFromFilePath(String path) in E:\CSharp\UnrealEngine\Plugins\source\Loader\ManagedLoadContext.cs:line 199
   at UnrealEngine.Runtime.Core.ManagedCommand(Command command) in D:\Work\Unity\UnrealCLR-master\UnrealCLR-master\Source\Managed\Runtime\Runtime.cs:line 267

Unreal Engine is 4.26.2
My proj file is:
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>net5.0</TargetFramework>
    <AppendTargetFrameworkToOutputPath>false</AppendTargetFrameworkToOutputPath>
  </PropertyGroup>

  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
    <OutputPath>D:\Documents\Unreal Projects\UnrealSamples\Managed\</OutputPath>
  </PropertyGroup>

  <ItemGroup>
    <Reference Include="UnrealEngine.Framework">
      <HintPath>..\UnrealCLR-master\UnrealCLR-master\Source\Managed\Framework\bin\Release\UnrealEngine.Framework.dll</HintPath>
    </Reference>
  </ItemGroup>

</Project>

**Comment by hmilyyang** [comment, marked as answer]
I reinstalled the UnrealCLR the next day, and the problem is gone, do not know why...
PS:
I found why. I delete the folder "ref" auto generated by visual studio in "Manage" foler, and problem is gone.

**Reply by nxrighthere** [reply]
You should [publish](https://docs.microsoft.com/en-us/dotnet/core/tutorials/publishing-with-visual-studio) your project to the managed folder to avoid such issues.
